krabs/krabs/provider.hpp

```diff
@@ -353,6 +396,15 @@ namespace krabs {
             for (auto &filter : filters_) {
                 filter.on_event(record);
             }
+
+            for (auto& callback : threadsafe_callbacks_) {
```

Comment:
Here and everywhere: * and & on the identifier.
```suggestion
            for (auto &callback : threadsafe_callbacks_) {
```